### PR TITLE
Quay Operator v3.3.0 Upgrade Procedure

### PR DESCRIPTION
**Pull request by kurtismullins**
This PR is my suggested procedure to update from Quay Operator v1.0.2 to v3.3.0.
It is _not_ written in the correct format, so please do not merge it.
This Pull Requests exists for review and/or re-writing purposes.

### CODE REVIEW

**Comment by kurtismullins** [comment]
Two notes from a quick glance over it myself:
1. The requirement for down-time and suggesting a maintenance window and informing users should be more prominently written towards the top of the documentation.
2. If the Quay deployment has any custom configuration they've performed by modifying the secret, or otherwise, then that configuration will have to be re-applied.

**Comment by chrisnegus** [comment]
@kurtismullins I copied this content into a separate section of the "Deploy Quay on OpenShift with Quay Operator" guide. I'm adding the comments from here into that doc.

File: modules/upgrade-steps.txt [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+			    INSTALL THE NEW OPERATOR
+-------------------------------------------------------------------------------
+
+Previously, Quay Operator v1.0.2 (and prior) were provided using the
+"community" Operator Hub catalog. In the latest release, the Quay Operator is
+released through official Red Hat channels.
+
+In the OpenShift UI, navigate to `Operators > OperatorHub` and then simply
+search for `Quay`. Ensure you are choosing the correct Quay Operator v3.3.0
+in the even that you encounter multiple, similar results. Simply click
```
[the remark this answers is not in the document]
Comment (chrisnegus) [reply modules/upgrade-steps.txt:156]: Good catch. Fixed in proc_deploy-quay-openshift-operator.adoc.

File: modules/upgrade-steps.txt [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+replicas is 1, so if one is not set than there is no work to be done here.
+
+
+-------------------------------------------------------------------------------
+			     UNINSTALL THE OPERATOR
+-------------------------------------------------------------------------------
+
+Verify that all Quay-related deployments and pods no longer exist within your
+namespace. Ensure that no other Quay deployments depend upon the installed
+Quay Operator v1.0.2 (or earlier).
```
[the remark this answers is not in the document]
Comment (chrisnegus) [reply modules/upgrade-steps.txt:140]: Good idea. Done.

File: modules/upgrade-steps.txt [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+				 UPDATE YOUR CR
+-------------------------------------------------------------------------------
+
+Ensure a backup is created of your original Custom Resource before making any
+changes.
+
+If your deployment does not specify any specific network-related configuration
+values, this step may not be necessary. Please refer to the documentation to
+ensure that your the configuration options in your current CR are still
+accurate for Quay Operator v3.3.0.
```
[the remark this answers is not in the document]
Comment (chrisnegus) [reply modules/upgrade-steps.txt:77]: I changed "ensure that your the" to "ensure that the". As for the CR, I'll leave this for @kurtismullins and keep it open for now.

File: modules/upgrade-steps.txt [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+information will aid them with the details needed to bring your cluster back
+to its original state. At minimum, the following information should be
+gathered:
+
+1. The Custom Resouce used to create your current Quay deployment.
+2. The output of running `oc get QuayEcosystem -o yaml > quayecosystem.yaml`
+   in your Project or Namespace.
+3. The hostnames currently used to access Quay, Clair, Quay's Config App,
+   Postgres, Redis, and Clair's Postgres instance. This can be achieved by
+   executing: `oc get routes -o yaml > old_routes.yaml`
```
[the remark this answers is not in the document]
Comment (chrisnegus) [reply modules/upgrade-steps.txt:52]: Done. Fixed in proc_deploy-quay-openshift-operator.adoc.

File: modules/upgrade-steps.txt [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+It is also recommended to perform a quick "smoke test" on your environment to
+ensure that the major functionality is working as expected. One example test
+may include performing pushes and pulls from the registry on existing, and new,
+images. Another example may be accessing Quay's UI as a registered user and
+ensuring that the expected TLS certificate is used. If you rely on the Quay
+Operator to generate a self-signed TLS certificate then keep in mind that a new
+certificate may have been created by this process.
+
+If multiple replicas are needed to scale your Quay registry, it is now safe
+to change the replica count to your desired quantity.
```
[the remark this answers is not in the document]
Comment (chrisnegus) [reply modules/upgrade-steps.txt:219]: Done. Fixed content in proc_deploy-quay-openshift-operator.adoc.

**Comment by chrisnegus** [comment]
I'm going to close this PR, since the content and corrections were added to the file proc_deploy-quay-openshift-operator.adoc in PR #56.

**Pull request closed without merging**